### ISSUE TITLE
The role comcast.sdkman is not idempotent

### ISSUE THREAD

**Issue by ollehallin** [Comcast/ansible-sdkman#48]
I have the following playbook:
```
---
- hosts: localhost
  connection: local
  vars:
  - sdkman_install_packages:
    - { candidate: java, version: '11.0.4-zulu' }
    - { candidate: java, version: '8.0.222-zulu' }
  - sdkman_defaults:
      java: '8.0.222-zulu'
  roles:
  - {role: comcast.sdkman, tags: sdkman}
```

Running the playbook twice fails the second time with the following play recap:
```
PLAY RECAP *****************************************************************************************************************
localhost                  : ok=10   changed=0    unreachable=0    failed=1    skipped=5    rescued=0    ignored=0
```

Expected play recap:
```
PLAY RECAP *****************************************************************************************************************
localhost                  : ok=11   changed=0    unreachable=0    failed=0    skipped=5    rescued=0    ignored=0
```

**comcast.sdkman version: 1.6.1**

**Comment by elliotweiser** [comment]
Which task failed? Hard to diagnose where the idempotence bug might be without knowing what failed.

**Comment by ollehallin** [comment]
```
TASK [comcast.sdkman : Install SDK candidates/versions] ***********************************************************************************************************************************************************************************
failed: [localhost] (item={'candidate': 'java', 'version': '11.0.5-zulu'}) => {"ansible_loop_var": "item", "changed": false, "cmd": ". /root/.sdkman/bin/sdkman-init.sh && sdk install java 11.0.5-zulu", "delta": "0:00:01.473126", "end": "2019-11-13 09:22:24.209065", "item": {"candidate": "java", "version": "11.0.5-zulu"}, "msg": "non-zero return code", "rc": 1, "start": "2019-11-13 09:22:22.735939", "stderr": "", "stderr_lines": [], "stdout": "\n\u001b[1;31mStop! java 11.0.5-zulu is already installed.\u001b[0m", "stdout_lines": ["", "\u001b[1;31mStop! java 11.0.5-zulu is already installed.\u001b[0m"]}
failed: [localhost] (item={'candidate': 'java', 'version': '8.0.222-zulu'}) => {"ansible_loop_var": "item", "changed": false, "cmd": ". /root/.sdkman/bin/sdkman-init.sh && sdk install java 8.0.222-zulu", "delta": "0:00:01.447681", "end": "2019-11-13 09:22:25.757027", "item": {"candidate": "java", "version": "8.0.222-zulu"}, "msg": "non-zero return code", "rc": 1, "start": "2019-11-13 09:22:24.309346", "stderr": "", "stderr_lines": [], "stdout": "\n\u001b[1;31mStop! java 8.0.222-zulu is already installed.\u001b[0m", "stdout_lines": ["", "\u001b[1;31mStop! java 8.0.222-zulu is already installed.\u001b[0m"]}
```

**Comment by ollehallin** [comment]
It seems like SDKMAN is returning non-zero the second time, which is not handled by the Ansible task.

**Comment by elliotweiser** [comment]
Good catch. The install block (and possibly uninstall) will need some defense because of how `sdk install` behaves for packages that are already installed.

https://github.com/Comcast/ansible-sdkman/blob/master/tasks/sdkman.yml#L27-L37

So I think this needs a `failed_when` which checks both the exit code and `changed_when` condition.

**Comment by elliotweiser** [comment]
Fixed in #51
